graphql_compiler/compiler/emit_sql.py

```diff
@@ -238,11 +363,18 @@ def _get_fold_outputs(self, fold_scope_location, join_descriptor, all_folded_out
                     # force column to have explicit label as opposed to anon_label
                     intermediate_fold_output_name = 'fold_output_' + fold_output.field
                     # add array aggregated output column to self._outputs
-                    self._outputs.append(
-                        sqlalchemy.func.array_agg(
-                            self.output_vertex_alias.c[fold_output.field]
-                        ).label(intermediate_fold_output_name)
-                    )
+                    if self._dialect == 'mssql':
+                        self._outputs.append(
+                            self._get_mssql_column(
+                                intermediate_fold_output_name,
+                                fold_output,
+                                self.join_info[-1]  # output vertex was most recently traversed
```

Comment:
How do we know `self.join_info` has anything in it? For example, I remember that `_construct_fold_joins()` pops everything off of that field — how do we know that function hasn't already been called?